src/main/kotlin/org/rust/lang/refactoring/extractFunction/RsExtractFunctionHandlerAction.kt

```diff
@@ -61,7 +62,12 @@ class RsExtractFunctionHandlerAction(
         stmt += if (firstParameter != null && firstParameter.name.endsWith("self") && firstParameter.type == null) {
             "self.${config.name}(${config.argumentsText})"
         } else {
-            val type = (config.containingFunction.owner as? RsFunctionOwner.Impl)?.impl?.typeReference?.text
+            val owner = config.containingFunction.owner
+            val type = when (owner) {
+                is RsFunctionOwner.Impl -> owner.impl.typeReference?.text
+                is RsFunctionOwner.Trait -> TyTypeParameter.Self.name
```

Comment:
I think it's better to use own "Self" literal instead of `TyTypeParameter.Self.name` to avoid unexpected changes while refactoring of `TyTypeParameter`